.github/workflows/kind-e2e-cosigned.yaml

```diff
@@ -0,0 +1,277 @@
+      run: |
+        echo '::group:: publish test image'
+        DIGEST=$(ko publish ./cmd/sample)
+        cat > pod.yaml <<EOF
+        apiVersion: v1
+        kind: Pod
+        metadata:
+          generateName: pod-test-
+        spec:
+          restartPolicy: Never
+          containers:
+          - name: sample
+            image: $DIGEST
+        EOF
+        cat > job.yaml <<EOF
+        apiVersion: batch/v1
+        kind: Job
+        metadata:
+          generateName: job-test-
+        spec:
+          template:
+            spec:
+              restartPolicy: Never
+              containers:
+              - name: sample
+                image: $DIGEST
+        EOF
+        echo '::endgroup::'
+
+        echo '::group:: test pod success (before labeling)'
+        # This time it should succeed!
+        if ! kubectl create -f pod.yaml ; then
+          echo Failed to create Pod in namespace without label!
+          exit 1
+        else
+          echo Successfully created Pod in namespace without label.
+        fi
+        echo '::endgroup::'
+
+        echo '::group:: test job success'
+        # This time it should succeed!
+        if ! kubectl create -f job.yaml ; then
+          echo Failed to create Job in namespace without label!
+          exit 1
+        else
+          echo Successfully created Job in namespace without label.
+        fi
+        echo '::endgroup::'
+
+        echo '::group:: enable verification'
+        kubectl label namespace default cosigned.sigstore.dev/include=true
+        echo '::endgroup::'
+
+        echo '::group:: test pod rejection'
+        if kubectl create -f pod.yaml ; then
+          echo Failed to block Pod creation!
+          exit 1
+        else
+          echo Successfully blocked Pod creation.
+        fi
+        echo '::endgroup::'
+
+        echo '::group:: test job rejection'
+        if kubectl create -f job.yaml ; then
+          echo Failed to block Job creation!
+          exit 1
+        else
+          echo Successfully blocked Job creation.
+        fi
+        echo '::endgroup::'
+
+        echo '::group:: sign test image'
+        cosign sign -key k8s://cosign-system/verification-key $DIGEST
+        echo '::endgroup::'
+
+        echo '::group:: test pod success'
+        # This time it should succeed!
+        if ! kubectl create -f pod.yaml ; then
+          echo Failed to create Pod with properly signed image!
+          exit 1
+        else
+          echo Successfully created Pod from signed image.
+        fi
+        echo '::endgroup::'
+
+        echo '::group:: test job success'
+        # This time it should succeed!
+        if ! kubectl create -f job.yaml ; then
+          echo Failed to create Job with properly signed image!
+          exit 1
+        else
+          echo Successfully created Job from signed image.
+        fi
+        echo '::endgroup::'
+
+
+    - name: Collect diagnostics
+      if: ${{ failure() }}
+      run: |
```

Comment:
Could we use scripts (potentially stored in a hack directory) instead of this long yaml file ?

Reply:
Perhaps I simple solution would have been to use `kind export logs` to a directory so it can be available as a bundle and download it.

Reply:
We can certainly change this to do that.  This snippet is just one I've been using across a variety of projects for this, and it's nice to be able to jump to directly the pod logs for the component you care about.

In Knative we uploaded the full logs to GCS, and they got very very large.  Not sure this will end up with as much logging or as many tests, but we can tweak this however we want 👍